docs/doc/14-sql-commands/20-query-syntax/02-query-with.md

```diff
@@ -37,6 +35,51 @@ WHERE  city = 'Montréal'
 ORDER  BY customername; 
 ```
 
+## Inline or Materialized?
+
+When using a CTE in a query, you can control whether the CTE is inline or materialized by using the MATERIALIZED keyword. Inlining means the CTE's definition is directly embedded within the main query, while materializing the CTE means calculating its result once and storing it in memory, reducing repetitive CTE execution.
+
+Suppose we have a table called *orders*, storing customer order information, including order number, customer ID, and order date. Now, we want to count the total number of orders for each customer and display only those customers with more than two orders.
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+  <TabItem value="inline" label="inline" default>
+
+In this query, the CTE customer_orders will be inlined during query execution. Databend will directly embed the definition of customer_orders within the main query to create a single execution plan.
+
+```sql
+WITH customer_orders AS (
+    SELECT customer_id, COUNT(*) AS order_count
+    FROM orders
+    GROUP BY customer_id
+)
+SELECT customer_id
+FROM customer_orders
+WHERE order_count > 2;
+```
+  </TabItem>
+  <TabItem value="MATERIALIZED" label="MATERIALIZED">
+
+In this case, we use the MATERIALIZED keyword, which means the CTE customer_orders will not be inlined. Instead, the CTE's result will be calculated and stored in memory during the CTE definition's execution. Later, when executing the main query, Databend will directly retrieve the pre-computed result from memory, avoiding the overhead of repeatedly executing the CTE.
```

Comment:
> avoiding the overhead of repeatedly executing the CTE.

If the example SQL uses the CTE more than once, it is only then that we can state this

Reply:
Updated to:  If a query uses a CTE more than once, this also helps avoid the overhead of repeatedly executing the CTE.